tools/clang/lib/CodeGen/CGHLSLMSFinishCodeGen.cpp

```diff
@@ -4034,12 +4034,17 @@ hlsl::DxilResourceProperties DxilObjectProperties::GetResource(llvm::Value *V) {
     return it->second;
   return DxilResourceProperties();
 }
-void DxilObjectProperties::updateGLC(llvm::Value *V) {
+void DxilObjectProperties::updateCoherence(llvm::Value *V,
+                                           bool updateGloballyCoherent,
+                                           bool updateReorderCoherent) {
   auto it = resMap.find(V);
   if (it == resMap.end())
     return;
 
-  it->second.Basic.IsGloballyCoherent ^= 1;
+  if (updateGloballyCoherent)
+    it->second.Basic.IsGloballyCoherent ^= 1;
+  if (updateReorderCoherent)
+    it->second.Basic.IsReorderCoherent ^= 1;
```

Comment:
This follows the existing pattern, but I'm not sure that the existing code was exactly what we should be doing, and the new flag introduces a new issue.  I've filed #7283 for follow up.

In my opinion, addressing this issue can wait until after merging this PR.